mobile/mysterium/proposals_manager.go

```diff
@@ -165,7 +174,16 @@ func (m *proposalsManager) addToCache(proposals []proposal.PricedServiceProposal
 	m.cachedAt = time.Now()
 }
 
-func (m *proposalsManager) getFromRepository(filter *proposal.Filter) ([]proposal.PricedServiceProposal, error) {
+func (m *proposalsManager) getFromRepository(req *GetProposalsRequest) ([]proposal.PricedServiceProposal, error) {
+	filter := req.toFilter()
+	if filter.NATCompatibility == "auto" {
+		natType, err := m.natProber.Probe(context.TODO())
```

Comment:
`context.TODO()` shows that something incomplete with it.

Reply:
That's right, getFromRepository function should also have contexts. Only top-level callers should use context.Background. I suspect contexts largely missing from our interfaces. For this reason we have hard to fix problem like this: #2004